flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManagerDriver.java

```diff
@@ -64,7 +65,7 @@
 
 	private final Time podCreationRetryInterval;
 
-	private final FlinkKubeClient kubeClient;
+	private FlinkKubeClient kubeClient;
```

Comment:
```suggestion
	private Optional<FlinkKubeClient> kubeClientOpt;
```
The client is used a lot in this driver. I would suggest to use `Optional` here to force a existence check whenever it is used.

Reply:
Actually, we should also have such protection for `podsWatch`. But it is not a problem of this PR.